postgresql/resource_postgresql_role.go

```diff
@@ -880,3 +915,20 @@ func alterSearchPath(txn *sql.Tx, d *schema.ResourceData) error {
 	}
 	return nil
 }
+
+func setStatementTimeout(txn *sql.Tx, d *schema.ResourceData) error {
+	if !d.HasChange(roleStatementTimeoutAttr) {
+		return nil
+	}
+
+	roleName := d.Get(roleNameAttr).(string)
+	statementTimeout := d.Get(roleStatementTimeoutAttr).(int)
+	sql := fmt.Sprintf(
+		"ALTER ROLE %s SET statement_timeout TO %d", pq.QuoteIdentifier(roleName), statementTimeout,
```

Comment:
You need to manage the case when `statement_timeout` is removed from the role TF config (so equal 0 I guess).
In this case we need to call:
```
ALTER ROLE xxx RESET statement_timeout ;
```
instead of setting to 0.

The reason behind that is that statement could be configured at a higher level (database, global configuration, etc...)
So if we remove `statement_attribute` from a role, we want to remove the custom value and use the higher one. Not set it to 0 (which will disable timeout for this role)